core/scripts/cli/voting/getResolvedVotesByRoundId.js

```diff
@@ -0,0 +1,76 @@
+const style = require("../textStyle");
+
+// Given a list of revealed votes for a user, return
+// the filtered list of votes that have successfully resolved a price
+// mapped to their round ID's.
+//
+// Cross reference revealed votes with a list of already-retrieved
+// rewards by the user to remove duplicate retrievals
+module.exports = async (web3, votingContract, account) => {
+  // All rewards available must be tied to formerly revealed votes
+  const revealedVotes = await votingContract.getPastEvents("VoteRevealed", {
+    filter: { voter: account },
+    fromBlock: 0
+  });
+
+  const resolvedVotesByRoundId = {};
+
+  for (let i = 0; i < revealedVotes.length; i++) {
+    const identifier = revealedVotes[i].args.identifier.toString();
+    const time = revealedVotes[i].args.time.toString();
+    const roundId = revealedVotes[i].args.roundId.toString();
+
+    try {
+      const price = (await votingContract.getPrice(identifier, time)).toString();
+
+      // If retrieveRewards returns 0, then the rewards have already been retrieved
+      let potentialRewards = await votingContract.retrieveRewards.call(account, roundId, [{ identifier, time }], {
+        from: account
+      });
+      potentialRewards = potentialRewards.toString();
+
+      if (potentialRewards !== "0") {
+        const resolvedVote = {
+          price,
+          name: `${web3.utils.hexToUtf8(identifier)} @ ${style.formatSecondsToUtc(time)}`,
+          identifier,
+          time,
+          roundId,
+          potentialRewards
+        };
+
+        // If this is a new roundId, begin a new array of resolved votes for the roundId
+        if (!resolvedVotesByRoundId[roundId]) {
+          resolvedVotesByRoundId[roundId] = [resolvedVote];
+        } else {
+          resolvedVotesByRoundId[roundId].push(resolvedVote);
+        }
+      } else {
+        // Account already retrieved this reward
+        continue;
+      }
+    } catch (err) {
+      // getPrice will throw if the vote has not resolved
+      continue;
+    }
+  }
+
+  // Create a mapping of round IDs to total rewards per round
+  let roundIds = [];
+  Object.keys(resolvedVotesByRoundId).forEach(id => {
+    let totalRewardsInRound = 0;
+    resolvedVotesByRoundId[id].forEach(reward => {
+      totalRewardsInRound += parseInt(reward.potentialRewards);
```

Comment:
nit: might be better to keep these as BNs, so you don't lose precision. You can always use the `fromWei()` function to do the division by `10e18`, and you can `.toString()` them whenever you need to print them out.

Reply:
good idea done